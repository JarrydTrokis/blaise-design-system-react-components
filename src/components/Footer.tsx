import React, {ReactElement} from "react";

function Footer(): ReactElement {
    return (
        <>
            <footer className="ons-footer">
                <div className="ons-footer__body" data-analytics="footer">
                    <div className="ons-container ">
                        <div className="ons-grid ons-grid--flex ons-grid--between">
                            <div className="ons-grid__col">
                                <div className="ons-u-mt-m ons-u-mb-s">
                                    <img src="https://ons-design-system.netlify.app/img/ons-logo-black-en.svg"
                                         alt="Office for National Statistics"/>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </footer>
        </>
    );
}

export default Footer;
