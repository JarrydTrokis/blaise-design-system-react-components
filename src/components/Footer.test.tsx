import React from "react";
import Enzyme, { render as enzymeRender, shallow } from "enzyme";
import { render } from "@testing-library/react";
import Adapter from "enzyme-adapter-react-16";
import Footer from "./Footer";

describe("Footer Test", () => {
    Enzyme.configure({ adapter: new Adapter() });

    it("matches Snapshot (using enzyme render)", () => {
        const wrapper = enzymeRender(<Footer />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render correctly", () => {
        const wrapper = shallow(<Footer />);
        expect(wrapper.exists()).toEqual(true);
    });

    it("should display 'Office for National Statistics' as part of the SVG", () => {
        const wrapper = render(<Footer />);
        expect(wrapper.getByTitle(/Office for National Statistics/)).toBeInTheDocument();
    });
});
