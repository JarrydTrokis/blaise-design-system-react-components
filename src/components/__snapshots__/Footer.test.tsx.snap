// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Test matches Snapshot 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "footer",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "footer__body",
          "data-analytics": "footer",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "container ",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "class": "grid grid--flex grid--between",
                },
                "children": Array [
                  Node {
                    "attribs": Object {
                      "class": "grid__col",
                    },
                    "children": Array [
                      Node {
                        "attribs": Object {
                          "class": "u-mt-m u-mb-s",
                        },
                        "children": Array [
                          Node {
                            "attribs": Object {
                              "alt": "Office for National Statistics",
                              "src": "https://ons-design-system.netlify.app/img/ons-logo-black-en.svg",
                            },
                            "children": Array [],
                            "name": "img",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "alt": undefined,
                              "src": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "alt": undefined,
                              "src": undefined,
                            },
                          },
                        ],
                        "name": "div",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                        },
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "data-analytics": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "data-analytics": undefined,
        },
      },
    ],
    "name": "footer",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
